### DISCUSSION TITLE
Opensource 5G RAN

### DISCUSSION THREAD

**Discussion by agr-priyanshu** [calee0219/awesome-5g#10] [General]
Do we have any Opensource 5G RAN implementation with CU-CU, CU-UP and DUs like free5gc?
The UERANSIM is great but it would be very helpful to have such a NF based setup.

**Comment by calee0219** [comment]
I'm not sure what do you means NF based setup. As my understanding, if you means deploy into container, UERANSIM has the ability due to it as a software simulator. But if you want to burn into hardware such as USRP, you may try OAI in 5G architecture.

**Reply by agr-priyanshu** [reply]
sorry, I wasn't very clear earlier.

What I meant is to have gNB segregated in to CU-UP, CU-CP and DU network functions; similar to Free5GC project which provides all Core network functions and that too slice aware? This would complete the E2E suite of configuration from provisioning perspective and would be a great addition.
